src/Microsoft.TestPlatform.Common/Filtering/Condition.cs

```diff
@@ -281,7 +273,95 @@ private string[] GetPropertyValue(Func<string, Object> propertyValueProvider)
             }
 
             return null;
-        }
+        }    
+
+        internal static IEnumerable<string> TokenizeFilterConditionString(string str)
+        {
+            if (str == null)
+            {
+                throw new ArgumentNullException(nameof(str));
+            }
+
+            return TokenizeFilterConditionStringWorker(str);
+
+            IEnumerable<string> TokenizeFilterConditionStringWorker(string s)
+            {
+                StringBuilder tokenBuilder = new StringBuilder();
+
+                var last = '\0';
+                for (int i = 0; i < s.Length; ++i)
+                {
+                    var current = s[i];
+
+                    if (last == FilterHelper.EscapeCharacter)
+                    {
+                        // Don't check if `current` is one of the special characters here.
+                        // Instead, we blindly let any character follows '\' pass though and 
+                        // relies on `FilterHelpers.Unescape` to report such errors.
+                        tokenBuilder.Append(current);
 
+                        if (current == FilterHelper.EscapeCharacter)
+                        {
+                            // We just encountered "\\" (escaped '\'), this will set last to '\0' 
+                            // so the next char will not be treated as a suffix of escape sequence.
```

Comment:
nit: treated as a **prefix** of escape sequence

Reply:
I think this should be "suffix". But there's definitely issue with my wording so I rewrote it :)